nntrainer/src/layers.cpp

```diff
@@ -451,19 +500,42 @@ int BatchNormalizationLayer::setOptimizer(Optimizer &opt) {
   this->opt.setOptParam(opt.getOptParam());
 
   this->epsilon = 0.0;
-  return this->opt.initialize(height, width, false);
+  return this->opt.initialize(dim.height(), dim.width(), false);
+}
+
+int BatchNormalizationLayer::setProperty(unsigned int key, const char *value) {
+  int status = ML_ERROR_NONE;
+
+  switch (static_cast<PropertyType>(key)) {
+  case PropertyType::input_shape:
+    status = dim.setTensorDim(value);
+    break;
+  case PropertyType::bias_zero: {
+    bool isTrue = (strcasecmp("true", value) == 0);
+    init_zero = isTrue;
+  } break;
+  case PropertyType::epsilon:
+    epsilon = std::stof(value);
+    break;
+  default:
+    ml_loge("Error: Unknown Layer Property Key");
+    status = ML_ERROR_INVALID_PARAMETER;
+    break;
+  }
+
+  return status;
 }
 
 Tensor BatchNormalizationLayer::forwarding(Tensor in, int &status) {
   Tensor temp;
 
   hidden = in;
 
-  mu = in.sum(0).multiply(1.0 / batch);
+  mu = in.sum(0).multiply(1.0 / dim.batch());
```

Comment:
How about adding `assert (dim.batch() > 0)` before division.

Reply:
fixed. By the way, layers are always checked the dimension during the initialization step. So it is guaranteed to be greater than 0. If the dimension is zero then it means this layer is not ready.